File: Leetcode/Arrays/977.squares-of-a-sorted-array.py
#
# @lc app=leetcode id=977 lang=python3
#
# [977] Squares of a Sorted Array
#

# @lc code=start
class Solution:
    def sortedSquares(self, A: List[int]) -> List[int]:
        for i in range(len(A)):
            if A[i] >= 0:
                break 
            else:
                A[i] *= -1 
        A.sort()
        for i in range(len(A)):
            A[i] = A[i] ** 2 

        return A
# @lc code=end

File: Leetcode/Practice/977.squares-of-a-sorted-array.cpp
/*
 * @lc app=leetcode id=977 lang=cpp
 *
 * [977] Squares of a Sorted Array
 */

// @lc code=start
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++) 
        {
            if(nums[i] >= 0)
            {
                break;
            }
            else
            {
                nums[i] *= -1;
            }
        }
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size(); i++)
        {
            nums[i] = nums[i] * nums[i];
        }
        return nums;
    }
};
// @lc code=end

